app/components/primer/underline_nav_component.rb

```diff
@@ -100,31 +117,38 @@ class UnderlineNavComponent < Primer::Component
     #
     # @param with_panel [Boolean] Whether the TabNav should navigate through pages or panels.
     # @param align [Symbol] <%= one_of(Primer::UnderlineNavComponent::ALIGN_OPTIONS) %> - Defaults to <%= Primer::UnderlineNavComponent::ALIGN_DEFAULT %>
+    # @param body_arguments [Hash] <%= link_to_system_arguments_docs %> for the body wrapper.
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-    def initialize(with_panel: false, align: ALIGN_DEFAULT, body_classes: "", **system_arguments)
+    def initialize(with_panel: false, align: ALIGN_DEFAULT, body_arguments: { tag: BODY_TAG_DEFAULT }, **system_arguments)
       @with_panel = with_panel
       @align = fetch_or_fallback(ALIGN_OPTIONS, align, ALIGN_DEFAULT)
 
       @system_arguments = system_arguments
       @system_arguments[:tag] = :nav
-      @system_arguments[:role] = :tablist
       @system_arguments[:classes] = class_names(
         @system_arguments[:classes],
         "UnderlineNav",
         "UnderlineNav--right" => @align == :right
       )
 
-      @body_arguments = {
-        tag: :div,
-        classes: class_names(
-          "UnderlineNav-body",
-          body_classes
-        )
-      }
+      @body_arguments = body_arguments
+      @body_tag = @body_arguments[:tag]&.to_sym
+
+      @body_arguments[:tag] = fetch_or_fallback(BODY_TAG_OPTIONS, @body_tag, BODY_TAG_DEFAULT)
+      @body_arguments[:classes] = class_names(
+        "UnderlineNav-body",
+        @body_arguments[:classes],
+        "list-style-none" => list?
+      )
+      @body_arguments[:role] = :tablist if with_panel
     end
 
     private
 
+    def list?
+      @body_tag == :ul
```

Comment:
What do you think about using `fetch_or_fallback` to define @body_tag? That way the value will be more predictable here.